### PR TITLE
handle dna/rna interactions by avoiding redundant DNA/RNA interactions

### PR DESCRIPTION
**Pull request by angelorosace**

### CODE REVIEW

File: structure_builder.py [hunk 1]
```diff
@@ -90,9 +90,12 @@ def superimpose_chain(chain, matching_chain, non_matching_chain):
 # Checks distance between all alpha carbons in the structure and chain atoms for
 # any clashes (where the two atoms are within a certain distance).
 def has_clashes_with_structure(structure, atoms, clash_distance=2):
-    is_alpha_carbon = lambda atom: atom.get_name() == 'CA'
-    structure_alpha_carbon_atoms = list(filter(is_alpha_carbon, structure.get_atoms()))
-    chain_alpha_carbon_atoms = list(filter(is_alpha_carbon, atoms))
+    # is_alpha_carbon = lambda atom: atom.get_name() == 'CA'
+    is_alpha_carbon_or_phosphate = lambda atom: atom.get_name() == 'CA' or atom.get_id() == "P"
```
Comment (iimry) [inline structure_builder.py]: Is there a difference between get_name and get_id?

File: structure_builder.py [hunk 2]
```diff
@@ -90,9 +90,12 @@ def superimpose_chain(chain, matching_chain, non_matching_chain):
...
+    # structure_alpha_carbon_atoms = list(filter(is_alpha_carbon, structure.get_atoms()))
+    structure_alpha_carbon_atoms = list(filter(is_alpha_carbon_or_phosphate, structure.get_atoms()))
+    # chain_alpha_carbon_atoms = list(filter(is_alpha_carbon, atoms))
```
Comment (iimry) [inline structure_builder.py]: You can just delete this stuff, we can always see this commit if for some reason we need it

File: structure_builder.py (hunk 1 above)
Comment (angelorosace) [reply structure_builder.py]: actually I don't think so, let me change to .get_name()

File: structure_builder.py (hunk 2 above)
Comment (angelorosace) [reply structure_builder.py]: oh yes...I forgot that there, thanks

**Pull request merged**